tailored-css.md

```diff
@@ -0,0 +1,120 @@
+tailored CSS.
+
+Our strategy leverages a tool
+called [Penthouse](https://github.com/pocketjoso/penthouse). It's a library that
+receives a CSS file and url, determines which styles are needed by the above the
+fold content, and outputs just those styles.
+
+## Multiple sizes and pages
+
+We realized quickly that we couldn't just take the results of a single
+Penthouse run and give it to users.
+
+The types of pages that we serve 90% of our traffic can be split up into 7
+different categories. Each has styles that aren't used at all by the others, so
+it made sense to generate a separate tailored CSS file for each.
+
+Our desktop site uses media queries to render slightly different styles for
+three ranges of screen widths. Because we can't know ahead of time which ones a
+user is going to need, our tailored CSS needs to include the styles for all of
+them.
+
+Even within a page category, a page can load in many different states. Different
+styles will be rendered above the fold as a result. For example, one of our
+grammar articles might have a custom table above the fold, but another might
+have a dialogue box.
+
+Our solution is to run Penthouse for each state, for each screen size, for each
+page category. The styles for each page category are then concatenated together
+and run through a series of postCSS plugins that deduplicate styles, organize
+media queries, and minify the files.
+
+## Increased complexity in development
+
+Though Penthouse is an amazing tool, we haven't found using it to be completely
+reliable. To be fair with the team behind it, most of that difficulty is
+probably the result of complexities in our own codebase.
+
+However, debugging when and why certain styles do or don't get included has not
+been an easy task. Because of that, we decided as a team that tailored CSS
+should be built at development time instead of at deploy time. Doing so means
+changes are reviewed as part of our normal pull request process.
+
+All in all, we run penthouse 43 times. The process takes around 5-6 minutes, and
+it affects our ability to iterate quickly.
+
+Tailored CSS also needs to be generated from *all* the most up to date styles.
+That means that before generating it, we have to pull and merge our master
+branch. When there are multiple pull requests open that change styles, we often
+end up having to the process multiple times before a PR can be merged.
+
+Generating production code based on what the site looks like in development
+introduces a dependency between the state of our development environment and
+production. To generate tailored CSS accurately, we now need to have running all the
+services that make up our site and recent database snapshot.
+
+## Moving forward
+
+Tailored CSS definitely got the job done, but it also generated a huge amount of
+technical debt. We are definitely open to other alternatives.
+
+One approach that I would love to take is to reorganize our UI code into
```

Comment:
I think it is generally agreed that we want to take this approach, it's just a question of how to get there. We'll do it in stages. Manual page category CSS is the first stage.